src/main/java/org/jpeek/plugin/JpeekMojo.java

```diff
@@ -23,25 +23,62 @@
  */
 package org.jpeek.plugin;
 
+import java.io.File;
+import java.io.IOException;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.jpeek.App;
 
 /**
  * {@link org.apache.maven.plugin.AbstractMojo} implementation for jPeek.
  *
  * @since 0.1
- * @todo #1:30min Implement jpeek plugin execution
- *  JpeekMojo must execute jpeek and analyze project files upon build.
- *  Use the jpeek jar from manve repository to execute it with in the
- *  project.
+ * @todo #3:30min Add some tests for this Mojo using AbstractMojoTestCase
+ *  from maven-plugin-testing-harness. A good resource for examples is
+ *  maven-checkstyle-plugin. It has been verified that it works from the
+ *  command line (see README).
+ * @todo #3:30min Add support for analyzing classes in the test directory.
+ *  This should output an analysis most certainly in a different directory
+ *  from the main classes.
  */
-@Mojo(name = "jpeek-analysis", defaultPhase = LifecyclePhase.TEST)
-public class JpeekMojo extends AbstractMojo {
+@Mojo(name = "analyze", defaultPhase = LifecyclePhase.SITE)
```

Comment:
@victornoel I think that it's better to bind the execution to `verify` phase, WDYT?

Reply:
@paulodamaso I checked the jacoco maven plugins, and it generates its report on verify, so it would be totally relevant yes I think, I will do it.